cdap/provider.go

```diff
@@ -75,9 +76,22 @@ func configureProvider(d *schema.ResourceData) (interface{}, error) {
 		return nil, err
 	}
 
-	return &Config{
+	c := &Config{
 		host:          d.Get("host").(string),
 		httpClient:    httpClient,
 		storageClient: storageClient,
-	}, nil
+	}
+	if err := healthcheck(c); err != nil {
+		return nil, fmt.Errorf("failed health check: %q, are the host and credentials valid?", err)
+	}
+	return c, nil
+}
+
+func healthcheck(c *Config) error {
+	req, err := http.NewRequest(http.MethodDelete, c.host, nil)
```

Comment:
What is this delete-ing?

Reply:
Good catch! This was a mistake, it should have been MethodGet.

But along the way I did some more testing and found a few interesting things.

You can make a call to a valid CDAP instance at the root ("/") endpoint and even without a valid token it will return without an error (empty body) including any method (get, delete, etc). That's why the original code despite calling MethodDelete didn't actually delete anything. So it can be useful for pinging if you want to just see if it exists there.

This also means a simple error check isn't sufficient to detect insufficient access (invalid token, etc). So instead I made it call GET on /v3/namespaces and treat the body as JSON.

Now if healthcheck passes we can be sure the instance URL is correct and the calling user has access :)